scipy/integrate/_ivp/common.py

```diff
@@ -107,6 +107,10 @@ def select_initial_step(fun, t0, y0, t_bound, max_step, f0, direction, order, rt
     if y0.size == 0:
         return np.inf
 
+    interval_length = abs(t_bound - t0)
+    if interval_length == 0.:
```

Comment:
Sorry, can we use 0.0 to denote floats? Not sure if it's a hard requirement by any Python linter, but I find it much nicer to read.